### PR TITLE
Warn if executable in bundle exec is empty

### PR DESCRIPTION
**Pull request by bronzdoc**
closes https://github.com/bundler/bundler/issues/5084

### CODE REVIEW

**Comment by segiddins** [comment]
👍 with a test, thanks!

**Comment by bronzdoc** [comment]
Done @segiddins

**Comment by bronzdoc** [comment]
Weird, all the tests pass in my local environment

**Comment by bronzdoc** [comment]
This is failing in ruby 1.8 :unamused:

File: spec/commands/exec_spec.rb [hunk 1]
```diff
@@ -514,6 +514,15 @@ def bin_path(a,b,c)
       end
     end
 
+    context "the executable is empty" do
+      let(:executable) { nil }
```
Comment (segiddins) [inline spec/commands/exec_spec.rb]: this should be an empty string instead of `nil`

File: lib/bundler/cli/exec.rb [hunk 2]
```diff
@@ -91,6 +91,12 @@ def ruby_shebang?(file)
         "#!/usr/bin/env jruby\n",
         "#!#{Gem.ruby}\n",
       ]
+
+      if File.zero?(file)
+        Bundler.ui.warn "#{file} is empty\n"
```
Comment (segiddins) [inline lib/bundler/cli/exec.rb]: the explicit `\n` shouldn't be necessary

**Comment by bronzdoc** [comment]
Kernel.exec in 1.8 is raising `Errno::ENOEXEC` if the file is empty even if set as an executable, I'll figure out something tonight.

**Comment by bronzdoc** [comment]
@segiddins I updated this PR let me know if this changes are ok with you

File: lib/bundler/cli/exec.rb [hunk 3]
```diff
@@ -86,11 +87,14 @@ def process_title(file, args)
     end
 
     def ruby_shebang?(file)
+      return false if File.zero?(file)
```
Comment (segiddins) [inline lib/bundler/cli/exec.rb]: this will cause us to check `File.zero?` twice for every file?

File: spec/commands/exec_spec.rb [hunk 4]
```diff
@@ -514,6 +514,19 @@ def bin_path(a,b,c)
...
     end
 
+    context "the executable is empty" do
+      let(:executable) { "" }
+
+      let(:exit_code) { 0 }
+      let(:expected) { "#{path} is empty" }
+      let(:expected_err) { "" }
+      if LessThanProc.with(RUBY_VERSION).call("1.9")
+        pending "Kernel.exec differ"
```
Comment (segiddins) [inline spec/commands/exec_spec.rb]: please add a comment explaining what you mean by this

Comment (bronzdoc) [inline spec/commands/exec_spec.rb]: will do

File: lib/bundler/cli/exec.rb (hunk 3 above)
Comment (bronzdoc) [inline lib/bundler/cli/exec.rb]: Yeah... I went back to the first implementation, sorry for the mess :grimacing:

**Comment by segiddins** [comment]
Thanks! @homu r+

**Pull request merged**